File: helfomat-api/src/main/java/de/helfenkannjeder/helfomat/api/organisation/event/ConfirmedChangeOrganizationEventDto.java
package de.helfenkannjeder.helfomat.api.organisation.event;

import de.helfenkannjeder.helfomat.core.organisation.OrganisationId;

import java.util.List;

/**
 * @author Valentin Zickner
 */
public class ConfirmedChangeOrganizationEventDto extends OrganizationEventDto {

    private String approvedBy;
    private String author;
    private String sources;
    private List<OrganizationEventDto> changes;

    private ConfirmedChangeOrganizationEventDto() {
    }

    public ConfirmedChangeOrganizationEventDto(OrganisationId organisationId, String approvedBy, String author, String sources, List<OrganizationEventDto> changes) {
        super(organisationId);
        this.approvedBy = approvedBy;
        this.author = author;
        this.sources = sources;
        this.changes = changes;
    }

    public String getApprovedBy() {
        return approvedBy;
    }

    public String getAuthor() {
        return author;
    }

    public String getSources() {
        return sources;
    }

    public List<OrganizationEventDto> getChanges() {
        return changes;
    }

    @Override
    public <T> T visit(OrganizationEventDtoVisitor<T> visitor) {
        return visitor.visit(this);
    }
}

File: helfomat-api/src/main/kotlin/de/helfenkannjeder/helfomat/api/organization/event/ConfirmedChangeOrganizationEventDto.kt
package de.helfenkannjeder.helfomat.api.organization.event

import de.helfenkannjeder.helfomat.core.organization.OrganizationId

/**
 * @author Valentin Zickner
 */
data class ConfirmedChangeOrganizationEventDto(
    override val organizationId: OrganizationId,
    val approvedBy: String,
    val author: String,
    val sources: String,
    val changes: List<OrganizationEventDto>,
    override val eventApplicable: Boolean
) : OrganizationEventDto {

    override fun <T> visit(visitor: OrganizationEventDtoVisitor<T>): T {
        return visitor.visit(this)
    }

}